Source/WebCore/loader/soup/ResourceLoaderSoup.cpp

```diff
@@ -33,12 +33,45 @@
 #include "ResourceError.h"
 #include "SharedBuffer.h"
 #include <gio/gio.h>
+#include <optional>
+#include <wtf/SortedArrayMap.h>
 #include <wtf/glib/GRefPtr.h>
 #include <wtf/glib/GUniquePtr.h>
 #include <wtf/glib/RunLoopSourcePriority.h>
+#include <wtf/text/ASCIILiteral.h>
+#include <wtf/text/WTFString.h>
 
 namespace WebCore {
 
+static std::optional<String> contentTypeLookUpForKnownResource(const char* filename)
+{
+    String fileName = String::fromLatin1(filename);
+
+    StringView extension;
+    size_t dotIndex = fileName.reverseFind('.');
+    if (dotIndex != notFound && dotIndex < (fileName.length() - 1))
+        extension = StringView(fileName).substring(dotIndex);
+
+    static constexpr std::pair<ComparableLettersLiteral, ASCIILiteral> pdfjsResourceMapping[] = {
```

Comment:
This is not used only for PDFjs, let's call it something else, for example `extensionTypeMapping`.

Reply:
We already have a better version of this in MIMETypeRegistry.cpp. Access it by calling `MIMETypeRegistry::mimeTypeForExtension`.